lib/redis_mutex.ex

```diff
@@ -1,90 +1,135 @@
 defmodule RedisMutex do
   @moduledoc """
-  An Elixir library for using Redis locks
+  An Elixir library for using Redis locks.
+  """
 
-  ## Setup
+  @type connection_options :: [
+          host: String.t(),
+          port: non_neg_integer(),
+          database: String.t() | non_neg_integer(),
+          username: String.t(),
+          password: Redix.password(),
+          timeout: timeout(),
+          sync_connect: boolean(),
+          exit_on_disconnection: boolean(),
+          backoff_initial: non_neg_integer(),
+          backoff_max: timeout(),
+          ssl: boolean(),
+          socket_opts: list(term()),
```

Comment:
is there a type for that we can point to?

Reply:
This list is taken directly from Redix's documentation

> `:socket_opts` (list of [`term/0`](https://hexdocs.pm/elixir/typespecs.html#built-in-types)) - specifies a list of options that are passed to the network layer when connecting to the Redis server. Some socket options (like `:active` or `:binary`) will be overridden by Redix so that it functions properly.

https://hexdocs.pm/redix/Redix.html#start_link/1-options